src/os/bluestore/zoned_types.h

```diff
@@ -0,0 +1,84 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+#ifndef CEPH_OS_BLUESTORE_ZONED_TYPES_H
+#define CEPH_OS_BLUESTORE_ZONED_TYPES_H
+
+#include "bluestore_types.h"
+#include "include/types.h"
+#include "kv/KeyValueDB.h"
+#include "os/kv.h"
+
+// Tracks two bits of information about the state of a zone: (1) number of dead
+// bytes in a zone and (2) the write pointer.  We represent these as a
+// two-element array of uint64_t because we use the existing RocksDB merge
+// operator, Int64ArrayMergeOperator, to update these.
+// 
+// We use the same struct for an on-disk and in-memory representation of the
+// state.  Calling persist method inserts the in-memory state to the database.
+struct zone_state_t {
+  uint64_t v[2];
+  uint64_t zone_num;
+  uint64_t zone_size;
```

Comment:
So zoned devices have zone size and zone capacity associated with each zone.  Zone size is fixed and it is some multiple of 2 and it is used to quickly figure out in which zone a given LBA falls.  Zone capacity, on the other hand, is the actual capacity of a zone and it determines how much data you can actually store in a zone.  With HM-SMR HDDs, zone size == zone capacity, but with ZNS SSDs, zone size >= zone capacity, and furthermore, zone capacity varies from zone to zone due to NAND flash characteristics.  So we do need to keep zone capacity per zone, but I will change the name from zone_size to zone_capacity because that is what it is.

Reply:
Ended up just removing zone_size, for now.